tests/pytests/unit/crypt/test_m2crypto.py

```diff
@@ -0,0 +1,160 @@
+]
+
+
+@pytest.mark.slow_test
+def test_gen_keys():
+    with patch.multiple(
+        os,
+        umask=MagicMock(),
+        chmod=MagicMock(),
+        access=MagicMock(return_value=True),
+    ):
+        with patch("M2Crypto.RSA.RSA.save_pem", MagicMock()) as save_pem:
+            with patch("M2Crypto.RSA.RSA.save_pub_key", MagicMock()) as save_pub:
+                with patch("os.path.isfile", return_value=True):
+                    assert crypt.gen_keys(
+                        "/keydir", "keyname", 2048
+                    ) == "/keydir{}keyname.pem".format(os.sep)
+                    save_pem.assert_not_called()
+                    save_pub.assert_not_called()
+
+                with patch("os.path.isfile", return_value=False):
+                    assert crypt.gen_keys(
+                        "/keydir", "keyname", 2048
+                    ) == "/keydir{}keyname.pem".format(os.sep)
+                    save_pem.assert_called_once_with(
+                        "/keydir{}keyname.pem".format(os.sep), cipher=None
+                    )
+                    save_pub.assert_called_once_with(
+                        "/keydir{}keyname.pub".format(os.sep)
+                    )
+
+
+@pytest.mark.slow_test
+def test_gen_keys_with_passphrase():
+    with patch.multiple(
+        os,
+        umask=MagicMock(),
+        chmod=MagicMock(),
+        chown=MagicMock(),
+        access=MagicMock(return_value=True),
+    ):
+        with patch("M2Crypto.RSA.RSA.save_pem", MagicMock()) as save_pem:
+            with patch("M2Crypto.RSA.RSA.save_pub_key", MagicMock()) as save_pub:
+                with patch("os.path.isfile", return_value=True):
+                    assert crypt.gen_keys(
+                        "/keydir", "keyname", 2048, passphrase="password"
+                    ) == "/keydir{}keyname.pem".format(os.sep)
+                    save_pem.assert_not_called()
+                    save_pub.assert_not_called()
+
+                with patch("os.path.isfile", return_value=False):
+                    assert crypt.gen_keys(
+                        "/keydir", "keyname", 2048, passphrase="password"
+                    ) == "/keydir{}keyname.pem".format(os.sep)
+                    callback = save_pem.call_args[1]["callback"]
+                    save_pem.assert_called_once_with(
+                        "/keydir{}keyname.pem".format(os.sep),
+                        cipher="des_ede3_cbc",
+                        callback=callback,
+                    )
+                    assert callback(None) == b"password"
+                    save_pub.assert_called_once_with(
+                        "/keydir{}keyname.pub".format(os.sep)
+                    )
+
+
+def test_sign_message(priv_key, msg, sig):
+    key = M2Crypto.RSA.load_key_string(salt.utils.stringutils.to_bytes(priv_key))
+    with patch("salt.crypt.get_rsa_key", return_value=key):
+        assert sig == salt.crypt.sign_message("/keydir/keyname.pem", msg)
```

Comment:
```suggestion
        assert salt.crypt.sign_message("/keydir/keyname.pem", msg) == sig
```
https://gist.github.com/s0undt3ch/92d065815f4ce526345f72d40a40074d